win32/src/job_object_helper.c

```diff
@@ -1,34 +1,171 @@
+    int result;
+
+    /*Codes_SRS_JOB_OBJECT_HELPER_18_035: [ If job_object_helper is NULL, job_object_helper_limit_memory shall fail and return a non-zero value. ]*/
+    /*Codes_SRS_JOB_OBJECT_HELPER_18_036: [ If percent_physical_memory is 0, job_object_helper_limit_memory shall fail and return a non-zero value. ]*/
+    /*Codes_SRS_JOB_OBJECT_HELPER_18_037: [ If percent_physical_memory is greater than 100, job_object_helper_limit_memory shall fail and return a non-zero value. ]*/
+    if (job_object_helper == NULL || percent_physical_memory == 0 || percent_physical_memory > 100)
+    {
+        LogError("Invalid arguments: job_object_helper = %p, percent_physical_memory = %" PRIu32 "", job_object_helper, percent_physical_memory);
+        result = MU_FAILURE;
+    }
+    else
+    {
+        /*Codes_SRS_JOB_OBJECT_HELPER_18_039: [ job_object_helper_limit_memory shall call SetInformationJobObject, passing JobObjectExtendedLimitInformation and a JOBOBJECT_EXTENDED_LIMIT_INFORMATION object with JOB_OBJECT_LIMIT_JOB_MEMORY set and JobMemoryLimit set to the percent_physical_memory percent of the physical memory in bytes. ]*/
+        JOBOBJECT_EXTENDED_LIMIT_INFORMATION extended_limit_information = {0};
+        extended_limit_information.BasicLimitInformation.LimitFlags  = JOB_OBJECT_LIMIT_JOB_MEMORY;
+        extended_limit_information.JobMemoryLimit = percent_physical_memory * job_object_helper->total_physical_memory / 100;
+        if (!SetInformationJobObject(job_object_helper->job_object, JobObjectExtendedLimitInformation, &extended_limit_information, sizeof(extended_limit_information)))
+        {
+            /*Codes_SRS_JOB_OBJECT_HELPER_18_041: [ If there are any failures, job_object_helper_limit_memory shall fail and return a non-zero value. ]*/
+            LogLastError("SetInformationJobObject failed");
+            result = MU_FAILURE;
+        }
+        else
+        {
+            /*Codes_SRS_JOB_OBJECT_HELPER_18_042: [ job_object_helper_limit_memory shall succeed and return 0. ]*/
+            result = 0;
+        }
+    }
+
+    return result;
 }
 
 IMPLEMENT_MOCKABLE_FUNCTION(, int, job_object_helper_limit_cpu, THANDLE(JOB_OBJECT_HELPER), job_object_helper, uint32_t, percent_cpu)
 {
-    (void) job_object_helper;
-    (void) percent_cpu;
-    return MU_FAILURE;
+    int result;
+
+    /*Codes_SRS_JOB_OBJECT_HELPER_18_043: [ If job_object_helper is NULL, job_object_helper_limit_cpu shall fail and return a non-zero value. ]*/
+    /*Codes_SRS_JOB_OBJECT_HELPER_18_044: [ If percent_cpu is 0, job_object_helper_limit_cpu shall fail and return a non-zero value. ]*/
+    /*Codes_SRS_JOB_OBJECT_HELPER_18_045: [ If percent_cpu is greater than 100, job_object_helper_limit_cpu shall fail and return a non-zero value. ]*/
+    if (job_object_helper == NULL || percent_cpu == 0 || percent_cpu > 100)
+    {
+        LogError("Invalid arguments: job_object_helper = %p, percent_cpu = %" PRIu32 "", job_object_helper, percent_cpu);
```

Comment:
> =  [](http://example.com/codeflow?start=54&length=3)

If at all possible, consider using the repo's established space before/after equal sign (or the lack of the space), which should keep the logs easier to parse and the code consistent.

If such a determination can be made quickly and easy.

Here and everywhere